packages/aws_common/test/logging/aws_logger_test.dart

```diff
@@ -220,5 +222,76 @@ void main() {
       expect(loggerPlugin.timesCalled, 5);
       expect(globalLoggerPlugin.timesCalled, 0);
     });
+
+    test('Get parent returns parent in the logger hierarchy', () {
+      final logger = AWSLogger().createChild('child').createChild('grandChild');
+      expect(logger.parent, AWSLogger().createChild('child'));
+      expect(logger.parent?.parent, AWSLogger());
+      expect(logger.parent?.parent?.parent, isNull);
+    });
+
+    test('Get children returns children in the logger hierarchy', () {
+      final logger = AWSLogger();
+      final childLogger1 = logger.createChild('child-1');
+      final childLogger2 = logger.createChild('child-2');
+      final grandChildLogger = childLogger1.createChild('grandChild');
+
+      expect(logger.children.contains(childLogger1), isTrue);
+      expect(logger.children.contains(childLogger2), isTrue);
+      expect(childLogger1.children.contains(grandChildLogger), isTrue);
+      expect(grandChildLogger.children, isEmpty);
+    });
+
+    test('getPlugin returns registered plugin in the logger hierarchy', () {
+      final loggerPlugin = MockLoggerPlugin();
+      final logger = AWSLogger()..registerPlugin(loggerPlugin);
+      expect(logger.getPlugin<MockLoggerPlugin>(), loggerPlugin);
+
+      final childLogger = logger.createChild('child');
+      expect(childLogger.getPlugin<MockLoggerPlugin>(), loggerPlugin);
+
+      childLogger.unregisterPlugin(loggerPlugin);
+      expect(childLogger.getPlugin<MockLoggerPlugin>(), loggerPlugin);
+      expect(logger.getPlugin<MockLoggerPlugin>(), loggerPlugin);
+
+      logger.unregisterPlugin(loggerPlugin);
+      expect(logger.getPlugin<MockLoggerPlugin>(), isNull);
+      expect(childLogger.getPlugin<MockLoggerPlugin>(), isNull);
+
+      childLogger.registerPlugin(loggerPlugin);
+      expect(logger.getPlugin<MockLoggerPlugin>(), isNull);
+      expect(childLogger.getPlugin<MockLoggerPlugin>(), loggerPlugin);
+    });
+
+    test(
+        'registerPlugin handles registering one plugin per type in the logger hierarchy',
+        () {
+      final plugin = MockLoggerPlugin();
+      final childLogger = AWSLogger()
+          .createChild('child')
+          .createChild('grandChild')
+          .createChild('grandGrandChild')
+        ..registerPlugin(plugin);
+      expect(childLogger.getPlugin<MockLoggerPlugin>(), plugin);
+
+      final newPlugin = MockLoggerPlugin();
+      AWSLogger().registerPlugin(newPlugin);
+      final parentPlugin = AWSLogger().getPlugin<MockLoggerPlugin>();
+      final childPlugin = childLogger.getPlugin<MockLoggerPlugin>();
+      expect(parentPlugin, newPlugin);
+      expect(childPlugin, newPlugin);
+      expect(newPlugin, isNot(plugin));
```

Comment:
I don't find it very intuitive that calling registerPlugin on a parent would deregister and close a plugin on the child.